encoder.go

```diff
@@ -127,9 +161,11 @@ func getFieldNameFromType(field reflect.StructField) (string, bool) {
 // - It will only take the first encoder.containerMaxSize elements of the struct
 // - If the field has a json tag it will become the field name
 // - Private fields and also values producing an error at encoding will be skipped
-func (encoder *encoder) encodeStruct(value reflect.Value, obj *wafObject, depth int) error {
+// - Even if the element values are invalid or null we still keep them to report the field name
+func (encoder *encoder) encodeStruct(value reflect.Value, obj *wafObject, depth int) {
 	if depth < 0 {
-		return errMaxDepth
+		encodeNative[uintptr](0, wafInvalidType, obj)
+		return
 	}
```

Comment:
Uhhhh negative depth sounds gross to me... I'd have made it unsigned and considered `0` as "tell me there's an object, but not what's in it".

Returning `wafInvalidType` here mis-represents the fact that the value is actually valid, but we have exceeded the encoding depth and are cutting it at this stage... which I don't love (troubleshooting will be more difficult).

Reply:
Fixed this here: https://github.com/DataDog/go-libddwaf/blob/e690a56d03cf815bf61978842aca8b311e47e910/encoder.go#L124